source/uwp/Renderer/lib/XamlBuilder.h

```diff
@@ -65,23 +65,67 @@ namespace AdaptiveCards::Rendering::Uwp
         void SetImageSource(T const& destination, winrt::ImageSource const& imageSource, winrt::Stretch stretch = winrt::Stretch::UniformToFill);
 
         template<typename T>
-        void SetImageOnUIElement(winrt::Uri const& imageUrl,
-                                 T const& uiElement,
-                                 winrt::AdaptiveCardResourceResolvers const& resolvers,
-                                 bool isAutoSize,
-                                 winrt::IInspectable const& parentElement,
-                                 winrt::IInspectable const& imageContainer,
-                                 bool isVisible,
-                                 bool isImageSvg = false,
-                                 winrt::Stretch stretch = winrt::Stretch::UniformToFill);
+        winrt::ImageSource SetImageOnUIElement(winrt::Uri const& imageUrl,
+                                               T const& uiElement,
+                                               winrt::AdaptiveCardResourceResolvers const& resolvers,
+                                               bool isAutoSize,
+                                               winrt::IInspectable const& parentElement,
+                                               winrt::IInspectable const& imageContainer,
+                                               bool isVisible,
+                                               bool isImageSvg = false,
+                                               winrt::Stretch stretch = winrt::Stretch::UniformToFill);
 
         winrt::ImageSource CreateImageSource(bool isImageSvg);
 
-        template<typename T> void PopulateImageFromUrlAsync(winrt::Uri const& imageUrl, T const& imageControl, bool const& isImageSvg);
+        template<typename T>
+        winrt::ImageSource PopulateImageFromUrlAsync(winrt::Uri const& imageUrl,
+                                                     T const& uiElement,
+                                                     bool isAutoSize,
+                                                     winrt::IInspectable const& parentElement,
+                                                     winrt::IInspectable const& imageContainer,
+                                                     bool isVisible,
+                                                     bool isImageSvg,
+                                                     winrt::Stretch stretch);
+
+        template<typename T, typename S>
```

Comment:
><typename T, typename S> [](http://example.com/codeflow?start=16&length=24)

When there are multiple templated things I think it is easier to understand them with more than one-character names.  In this case perhaps it would be `<typename TElement, typename TStream>`?